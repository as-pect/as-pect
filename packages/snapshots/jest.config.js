module.exports = {
  preset: 'ts-jest',
  testEnvironment: 'node',
  collectCoverage: false,
  testMatch: ["**/__tests__/**/*.spec.[jt]s"],
  testPathIgnorePatterns: ["/assembly/", "/node_modules/"],
};
