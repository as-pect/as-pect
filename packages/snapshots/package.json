{
  "name": "@as-pect/snapshots",
  "version": "5.0.1",
  "description": "A package for dealing with snapshots",
  "main": "lib/index.js",
  "scripts": {
    "nearley": "nearleyc src/parser/grammar.ne -o src/parser/grammar.ts",
    "rimraf": "rimraf lib",
    "test": "jest",
    "test:ci": "jest",
    "test:update-snapshots": "jest -u",
    "tsc:all": "run-s rimraf nearley tsc:default tsc:amd",
    "tsc:amd": "tsc --outfile lib/as-pect.core.amd.js --module amd --declaration --declarationMap",
    "tsc:default": "tsc"
  },
  "keywords": [
    "snapshots",
    "testing",
    "diff"
  ],
  "author": "Joshua Tenner <tenner.joshua@gmail.com>",
  "license": "MIT",
  "devDependencies": {
    "@types/nearley": "^2.11.1",
    "jest": "^26.6.3",
    "ts-jest": "^26.5.0"
  },
  "dependencies": {
    "diff": "^5.0.0",
    "nearley": "^2.20.1"
  }
}
