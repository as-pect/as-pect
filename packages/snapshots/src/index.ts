export * from "./Snapshot";
export * from "./SnapshotDiff";
export * from "./SnapshotDiffResult";
