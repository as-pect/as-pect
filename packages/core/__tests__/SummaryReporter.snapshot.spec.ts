import { SummaryReporterWrapper } from "./setup/SummaryReporterWrapper.js";
import { TestContext } from "../src/index.js";
import { promises as fs } from "fs";
import { instantiate } from "@assemblyscript/loader";
import { Snapshot } from "@as-pect/snapshots";

test("snapshots", async () => {
  const binary = await fs.readFile("./assembly/jest-reporter-snapshot.wasm");
  const reporter = new SummaryReporterWrapper();
  const ctx = new TestContext({
    binary,
    reporter,
    fileName: "./assembly/jest-reporter-snapshot.ts",
    snapshots: Snapshot.from(
      new Map<string, string>([
        // modified
        [
          "a few snapshots!~first vec3[0]",
          "Vec3 {\n  x: 1.0,\n  y: 2.0,\n  z: 4.0,\n  a: 10.0\n}",
        ],
        // same
        ["a few snapshots!~a string[0]", '"some string"'],
        // added: ['a few snapshots!~some integer[0]', '504'],
        ["a few snapshots!~some float value[0]", "3.14"],
        // removed from module
        ["this!~is!~removed", "false"],
      ]),
    ),
  });
  ctx.run(await instantiate(binary, ctx.createImports()));
  for (const [name, values] of reporter.snapshots.entries()) {
    for (const value of values) {
      expect(value).toMatchSnapshot(name);
    }
  }
});
