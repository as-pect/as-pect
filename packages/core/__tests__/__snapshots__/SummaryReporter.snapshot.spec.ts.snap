// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots: onFinish ./assembly/jest-reporter-snapshot.ts 1`] = `
"‚ùå ./assembly/jest-reporter-snapshot.ts Pass: 1 / 1 Todo: 0 Time: 0ms
  Failed: logs
     [Log]: \\"Hello world!\\"
     [Log]: 42
     [Log]: <Vec3>null
     [Log]: Vec3 {
                x: 1.0,
                y: 2.0,
                z: 3.0
            }
     [Log]: Array<i32> [
                1,
                2,
                3
            ]
     [Log]: 9999999999999 as i64
     [Log]: 1234567 as u32
     [Log]: 999999999999 as u64
     [Log]: -1 as i8
     [Log]: -1 as i16
     [Log]: true
     [Log]: false
     [Log]: Uint8Array [ 0, 1, 2 ]
     [Log]: Int8Array [ -1, -2, -3 ]
[Snapshot]: a few snapshots!~first vec3[0]
  Vec3 {
    x: 1.0,
    y: 2.0,
-   z: 4.0,
-   a: 10.0
+   z: 3.0
  }

[Snapshot]: a few snapshots!~some integer[0]
+ 504

[Snapshot]: this!~is!~removed
- false

"
`;
